plugins/CorePluginsAdmin/Controller.php

```diff
@@ -367,11 +374,13 @@ public function activate($redirectAfter = true)
 
     public function deactivate($redirectAfter = true)
     {
-        $pluginName = $this->initPluginModification(static::DEACTIVATE_NONCE);
-        $this->dieIfPluginsAdminIsDisabled();
-
-        $this->pluginManager->deactivatePlugin($pluginName);
-        $this->redirectAfterModification($redirectAfter);
+        if($this->isAllowedToTroubleshootAsSuperUser()) {
+            Piwik::doAsSuperUser(function() use ($redirectAfter) {
```

Comment:
as a result of this code, anyone who knows the secret `salt` of a Piwik instance may be able to deactivate plugins, as long as they also know the Nonce so it should be very safe